### PR TITLE
Major package update

### LINKED ISSUE #18: Velocity/Baz left as np.inf or np.nan
Noticed with some datasets, values for velocity and baz are often "calculated" as inf or nan when mdccm is high . I believe this happens in this fast_lts_array.py script - several variables are initiated as np.inf or np.nan and they make it through the processing without being reassigned.

### PR DESCRIPTION
**Pull request by jwbishop**
Major Package Update

1) Package internals re-written in an object-oriented framework that uses [numba](https://numba.readthedocs.io/en/stable/) for improved computational speed.
2) Uncertainty estimates for back-azimuth and trace-velocity estimates are now calculated automatically using the slowness ellipse method of [Szuberla and Olson](https://asa.scitation.org/doi/10.1121/1.1635407).
3) Bug Fixes: Fixes #18, #19, #20, #22
4) Documentation Updates. User notes are now included.

### CODE REVIEW

**Comment by liamtoney** [comment]
@jwbishop can you summarize what the breaking changes are here? Looking at the documentation [here](https://uaf-lts-array.readthedocs.io/en/master/api/lts_array.ltsva.html) it looks like `ltsva()` function inputs/outputs are unchanged — so I can drop this new code into my existing workflows?

**Comment by liamtoney** [comment]
> Looking at the documentation [here](https://uaf-lts-array.readthedocs.io/en/master/api/lts_array.ltsva.html) it looks like `ltsva()` function inputs/outputs are unchanged — so I can drop this new code into my existing workflows?

Hmmm, are those docs not up to date? I looked at the new docstring after pulling the code and it looks like `rij` has been deprecated in favor of lists of lat and lon?

**Comment by liamtoney** [comment]
> Hmmm, are those docs not up to date? I looked at the new docstring after pulling the code and it looks like rij has been deprecated in favor of lists of lat and lon?

I made an issue: #27

**Comment by jwbishop** [comment]
Hi @liamtoney, thanks for this question. Setup instructions and documentation generation have now been fixed by #28.

There are two key changes that users need to know for this update!
1) Inputs: Latitude and Longitude values are now the preferred inputs. The code will now internally project onto a local planar coordinate system with a [vincety inverse](https://docs.obspy.org/packages/autogen/obspy.geodetics.base.calc_vincenty_inverse.html).
2) The code now automatically outputs 90% confidence intervals for the trace velocity and back-azimuth, so 2 additional outputs need to be accounted for in current workflows.

**Comment by jwbishop** [comment]
The code input/output is flexible, so I am open to user feedback.

**Comment by jwbishop** [comment]
Another note: the least squares estimation (for both trimmed and ordinary least squares) is now largely completed by internal codes compiled by `Numba`. This results in a massive speedup of the computational time, but it's unclear to me if I can output completion percentages to the external device at the same time. The most expensive part of the code is now the cross-correlations, and output info can be easily added here if users are interested.

**Comment by liamtoney** [comment]
> it's unclear to me if I can output completion percentages to the external device at the same time

Can you explain what you mean by "external device" here?
